### PR TITLE
attempts to make reversible reaction deletion work

### PR DESCRIPTION
**Pull request by pstjohn**
Some in-progress attempts at fixing `model.remove_reactions` and `model._populate_solver`. Still not working.

Addresses #655

### CODE REVIEW

**Comment by cdiener** [comment]
The problem is at https://github.com/opencobra/cobrapy/blob/76ed164d4fada06395a8fc9be80e6bedde825c8e/cobra/core/model.py#L613 . This will already trigger a re-add of the variables on context exit. But `_populate_solver` is also called on exit and also adds the variables. Maybe adding `_populate_solver` to the context *before*  the call to `remove_cons_vars` works.

**Comment by pstjohn** [comment]
Thanks for the tip! This works locally, lets see what CI says.

**Comment by Midnighter** [comment]
@pstjohn are the calls to `self.solver.update()` strictly necessary? I'm asking because updating the solver after every change defeats the purpose of collecting them and then doing batch modifications.

**Comment by pstjohn** [comment]
Doesn't look like it, good call.

**Comment by pstjohn** [comment]
@Midnighter, this is RTM from my end. Any other thoughts?

File: cobra/core/model.py [hunk 1]
```diff
@@ -745,16 +747,24 @@ def _populate_solver(self, reaction_list, metabolite_list=None):
...
+                    reaction.id, lb=forward_lb, ub=forward_ub)
+                reverse_variable = self.problem.Variable(
+                    reaction.reverse_id, lb=reverse_lb, ub=reverse_ub)
 
-            forward_variable = self.problem.Variable(
-                reaction.id, lb=forward_lb, ub=forward_ub)
-            reverse_variable = self.problem.Variable(
-                reaction.reverse_id, lb=reverse_lb, ub=reverse_ub)
+                self.add_cons_vars([forward_variable, reverse_variable])
+                self.solver.update()
```
Comment (Midnighter) [inline cobra/core/model.py]: This one is necessary?

**Comment by Midnighter** [comment]
If you can comment on that one line, I'm happy to merge.

**Comment by pstjohn** [comment]
Oh, sorry, thought you meant the context'ed model.solver.update. No, I don't think we need that line. I'll delete it

**Comment by pstjohn** [comment]
should fix #661 as well

**Pull request merged**